atlasaction/action_test.go

```diff
@@ -87,6 +93,234 @@ func TestMigrateApply(t *testing.T) {
+			case strings.Contains(query.Query, "pushDir"):
+				pushDirCalls.inc()
+				require.NoError(t, json.Unmarshal(query.Variables, &query.PushDir))
+				require.Equal(t, query.PushDir.Tag, "sha1234")
+				require.Equal(t, query.PushDir.Slug, "test-dir")
+				fmt.Fprint(w, `{"data":{"pushDir":{"url":"https://some-org.atlasgo.cloud/dirs/314159/tags/12345"}}}`)
+			}
+		}))
+		tt.setupConfigWithLogin(t)
+		tt.setInput("dir", "file://testdata/migrations")
+		tt.setInput("dir-name", "test-dir")
+		tt.setInput("dev-url", "sqlite://file?mode=memory")
+		tt.env["GITHUB_REPOSITORY"] = "repository"
+		tt.env["GITHUB_REF_NAME"] = "testing-branch"
+		tt.env["GITHUB_SHA"] = "sha1234"
+		err := MigratePush(context.Background(), tt.cli, tt.act)
+		require.NoError(t, err)
+		require.Equal(t, syncDirCalls.get(), int32(1))
+		require.Equal(t, pushDirCalls.get(), int32(1))
+		require.NoError(t, err)
+		outputs, _ := tt.outputs()
+		url := outputs["url"]
+		require.Equal(t, "https://some-org.atlasgo.cloud/dirs/314159/tags/12345", url)
+	})
+}
+
+type (
+	httpTest struct {
+		*test
+		srv        *httptest.Server
+		cloudToken string
+	}
+
+	pushDir struct {
+		Slug   string `json:"slug"`
+		Tag    string `json:"tag"`
+		Driver string `json:"driver"`
+		Dir    string `json:"dir"`
+	}
+	syncDir struct {
+		Slug    string        `json:"slug"`
+		Driver  string        `json:"driver"`
+		Dir     string        `json:"dir"`
+		Context *ContextInput `json:"context"`
+	}
+
+	graphQLQuery struct {
+		Query     string          `json:"query"`
+		Variables json.RawMessage `json:"variables"`
+		PushDir   struct {
+			pushDir `json:"input"`
+		}
+		SyncDir struct {
+			syncDir `json:"input"`
+		}
+	}
+)
+
+type atomicInt int32
+
+func (c *atomicInt) inc() int32 {
+	return atomic.AddInt32((*int32)(c), 1)
+}
+
+func (c *atomicInt) get() int32 {
+	return atomic.LoadInt32((*int32)(c))
+}
+
+func newHttpT(t *testing.T, handler http.HandlerFunc) *httpTest {
+	tt := &httpTest{
+		test:       newT(t),
+		cloudToken: "123456789",
+	}
+	tt.srv = fakeCloud(t, tt, handler)
+	return tt
+}
+
+func fakeCloud(t *testing.T, tt *httpTest, handler http.HandlerFunc) *httptest.Server {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(t, "Bearer "+tt.cloudToken, r.Header.Get("Authorization"))
+		if handler != nil {
+			handler(w, r)
+		}
+	}))
+	t.Cleanup(func() {
+		srv.Close()
+	})
+	return srv
+}
+
+func (tt *httpTest) setupConfigWithLogin(t *testing.T) {
+	tt.setInput("config", generateHCL(tt.srv.URL, tt.cloudToken, t))
+}
+
+func generateHCL(url, token string, t *testing.T) string {
+	tmpl := `
+	atlas {
+		cloud {
+			token = "{{ .Token }}"
+		{{- if .URL }}
+			url = "{{ .URL }}"
+		{{- end }}
+		}	  
+	}
+	env "test" {
+		url = "sqlite://file?mode=memory"
+		dev = "sqlite://file?mode=memory"
+		migration  {
+			dir = "file://testdata/migrations"
+		}
+	  }
+	`
+	config := template.Must(template.New("atlashcl").Parse(tmpl))
```

Comment:
```suggestion
	config := template.Must(template.New("atlashcl").Parse(`
atlas {
	cloud {
		token = "{{ .Token }}"
	{{- if .URL }}
		url = "{{ .URL }}"
	{{- end }}
	}	  
}
env "test" {
	url = "sqlite://file?mode=memory"
	dev = "sqlite://dev?mode=memory"
	migration  {
		dir = "file://testdata/migrations"
	}
}`))
```
You might want to pass the dir as variable or just not define it in this file